### PR TITLE
[docs] Add docs for reworked Airbyte Cloud API

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

As title.

## How I Tested These Changes

Docs preview

### CODE REVIEW

File: docs/content/integrations/airbyte/airbyte-cloud.mdx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+- How to customize how Airbyte Cloud connection tables are materialized.
+
+<details>
+  <summary>Prerequisites</summary>
+
+- The `dagster` and `dagster-airbyte` libraries installed in your environment
+- Familiarity with asset definitions and the Dagster asset graph
+- Familiarity with Dagster resources
+- Familiarity with Airbyte Cloud concepts, like connections and connection tables
+- A Airbyte Cloud workspace
```
Comment (neverett) [inline docs/content/integrations/airbyte/airbyte-cloud.mdx:24]:
```suggestion
- An Airbyte Cloud workspace
```

File: docs/content/integrations/airbyte/airbyte-cloud.mdx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+- A Airbyte Cloud client ID and client secret. For more information, see [Configuring API Access](https://docs.airbyte.com/using-airbyte/configuring-api-access) in the Airbyte Cloud REST API documentation.
```
Comment (neverett) [inline docs/content/integrations/airbyte/airbyte-cloud.mdx:25]:
```suggestion
- An Airbyte Cloud client ID and client secret. For more information, see [Configuring API Access](https://docs.airbyte.com/using-airbyte/configuring-api-access) in the Airbyte Cloud REST API documentation.
```

**Review by neverett — CHANGES_REQUESTED** [review]
Left some small copyediting suggestions, otherwise LGTM

File: docs/content/integrations/airbyte/airbyte-cloud.mdx (hunk 1 above)
Comment (maximearmstrong) [reply docs/content/integrations/airbyte/airbyte-cloud.mdx]: Done in [ba8ac3a](https://github.com/dagster-io/dagster/pull/26595/commits/ba8ac3a7e33894c083aa2056ae17766afe62a305)

File: docs/content/integrations/airbyte/airbyte-cloud.mdx (hunk 2 above)
Comment (maximearmstrong) [reply docs/content/integrations/airbyte/airbyte-cloud.mdx]: Done in [ba8ac3a](https://github.com/dagster-io/dagster/pull/26595/commits/ba8ac3a7e33894c083aa2056ae17766afe62a305)

**Review by neverett — APPROVED** [review]

**Pull request merged**